zetaclient/orchestrator/bootstrap.go

```diff
@@ -0,0 +1,343 @@
+package orchestrator
+
+import (
+	"context"
+	"fmt"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/pkg/errors"
+
+	"github.com/zeta-chain/zetacore/zetaclient/chains/base"
+	btcobserver "github.com/zeta-chain/zetacore/zetaclient/chains/bitcoin/observer"
+	"github.com/zeta-chain/zetacore/zetaclient/chains/bitcoin/rpc"
+	btcsigner "github.com/zeta-chain/zetacore/zetaclient/chains/bitcoin/signer"
+	evmobserver "github.com/zeta-chain/zetacore/zetaclient/chains/evm/observer"
+	evmsigner "github.com/zeta-chain/zetacore/zetaclient/chains/evm/signer"
+	"github.com/zeta-chain/zetacore/zetaclient/chains/interfaces"
+	"github.com/zeta-chain/zetacore/zetaclient/config"
+	zctx "github.com/zeta-chain/zetacore/zetaclient/context"
+	"github.com/zeta-chain/zetacore/zetaclient/db"
+	"github.com/zeta-chain/zetacore/zetaclient/metrics"
+)
+
+// backwards compatibility
+const btcDatabaseFilename = "btc_chain_client"
+
+// CreateSignerMap creates a map of interfaces.ChainSigner (by chainID) for all chains in the config.
+// Note that signer construction failure for a chain does not prevent the creation of signers for other chains.
+func CreateSignerMap(
+	ctx context.Context,
+	tss interfaces.TSSSigner,
+	logger base.Logger,
+	ts *metrics.TelemetryServer,
+) (map[int64]interfaces.ChainSigner, error) {
+	signers := make(map[int64]interfaces.ChainSigner)
+	_, _, err := syncSignerMap(ctx, tss, logger, ts, &signers)
+	if err != nil {
+		return nil, err
+	}
+
+	return signers, nil
+}
+
+// syncSignerMap synchronizes the given signers map with the signers for all chains in the config.
+// This semantic is used to allow dynamic updates to the signers map.
+func syncSignerMap(
+	ctx context.Context,
+	tss interfaces.TSSSigner,
+	logger base.Logger,
+	ts *metrics.TelemetryServer,
+	signers *map[int64]interfaces.ChainSigner,
+) (int, int, error) {
+	if signers == nil {
+		return 0, 0, errors.New("signers map is nil")
+	}
+
+	app, err := zctx.FromContext(ctx)
+	if err != nil {
+		return 0, 0, errors.Wrapf(err, "failed to get app context")
+	}
+
+	var (
+		added, removed int
+
+		presentChainIDs = make([]int64, 0)
+
+		onAfterSet = func(chainID int64, _ interfaces.ChainSigner) {
```

Comment:
Can we reuse added/removed terminology here instead of set/unset?

Reply:
https://github.com/zeta-chain/node/pull/2497/commits/2f06e5639910af4fbfeb4261ff0faf4a86b53af1